www/source/blog/2019-03-04-introducing-the-install-hook.html.md

```diff
@@ -0,0 +1,124 @@
+
+## Why a new hook?
+
+Sometimes installing software involves much more than simply laying files on disk. In many cases, simply extracting the files packaged in a `hart` file to the `/hab/pkgs` folders and ensuring that some environment variables are set at runtime are all that is needed for an application to behave. However in some cases, there may be required system API calls needed to ensure that software components are correctly registered with certain OS sub systems. This is often the case with Windows applications which may require registry keys to be set, Windows features to be enabled or complicated install scripts and binaries to run before the software can be considered installed and ready to use.
+
+Today we manage these kinds of applications in Habitat by including this installation behavior inside of an `init` or `run` hook. This has worked but what if the application is a command line tool, a library, or a stand alone application that runs outside of a Supervisor? In such cases, you must deploy the application as a service and include a `run` hook that simply spins and sleeps. Even if the application is a service, what if it has dependencies like IIS or COM components that require the execution of scripts to fully register and configure those key dependencies. Well we either bundle those dependencies with the parent app or include their installation scripts with the parent package requiring the parent application to "know" intimate details about its dependencies.
+
+With the `install` hook we can now break these applications up so that any installation behavior needed in a dependant package can be managed by that package. This also means that one can install Habitat packages that run as stand-alone applications and not be required to run them as services in order to execute their installation scripts.
+
+## An example install hook scenario
+
+To illustrate where an `install` hook may be convenient, I'll use a [legacy Windows web application](https://github.com/habitat-sh/sqlwebadmin) that we use in some Habitat demonstrations. This application requires IIS and ASP.Net 3.5 as well as a dependency on a COM library. Its `pkg_deps` looks like:
+
+```
+$pkg_deps=@("core/dsc-core", "core/sql-dmo")
+```
+
+Its `init` hook registers the COM component packaged in the `sql-dmo` package:
+
+```
+."$env:SystemRoot\SysWow64\regsvr32.exe" /s "{{pkgPathFor "core/sql-dmo"}}\Program Files (x86)\Microsoft SQL Server\80\Tools\Binn\sqldmo.dll"
```

Comment:
Is "Binn" correct, or should it be "Bin"?

Reply:
yeah `binn` its fairly common for MS software